src/ajax.js

```diff
@@ -13,6 +13,7 @@ define([
 
 var
 	rhash = /#.*$/,
+	rextraQuestions = /\?+/,
```

Comment:
Well, to be honest, I think we shouldn't try and fix anything because it is working as intended. The original bug report is bogus in that it is asking to cater to an illformed url.